code/modules/organs/external/tail.dm

```diff
@@ -0,0 +1,27 @@
+/obj/item/organ/external/tail
+	var/tail_icon
+	var/tail_state
+	var/tail_hair
+	var/tail_animation
+
+	organ_tag = BP_TAIL
+	render_alpha = 0
+	name = "tail"
+	max_damage = 20
+	min_broken_damage = 10
+	w_class = ITEM_SIZE_NORMAL
+	body_part = SLOT_LOWER_BODY
+	parent_organ = BP_GROIN
+	joint = "tail"
+	amputation_point = "tail"
+	artery_name = "vein"
+	arterial_bleed_severity = 03
+	limb_flags = ORGAN_FLAG_CAN_AMPUTATE | ORGAN_FLAG_CAN_BREAK
+
+/obj/item/organ/external/tail/set_dna(datum/dna/new_dna)
+	. = ..()
+	if(owner.bodytype || !(tail_icon && tail_state && tail_hair && tail_animation))//no overriding if something there
```

Comment:
Why do you store these values on the tail anyway? Why not just grab them from `owner.bodytype` when you are generating the icon?

Reply:
The tail should not adjust to the body during surgery. So, data on its appearance should be saved in it.